impl/src/main/java/com/okta/sdk/impl/oauth2/AccessTokenRetrieverServiceImpl.java

```diff
@@ -0,0 +1,90 @@
+package com.okta.sdk.impl.oauth2;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.okta.commons.http.HttpException;
+import com.okta.commons.http.HttpMethod;
+import com.okta.commons.lang.Assert;
+import com.okta.commons.lang.Strings;
+import com.okta.sdk.impl.config.ClientConfiguration;
+import com.okta.sdk.oauth2.ErrorResponse;
+import com.okta.sdk.oauth2.OAuth2AccessToken;
+import okhttp3.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+
+import static com.okta.sdk.impl.util.OAuth2Utils.createSignedJWT;
+
+public class AccessTokenRetrieverServiceImpl implements AccessTokenRetrieverService {
+    private static final Logger log = LoggerFactory.getLogger(AccessTokenRetrieverServiceImpl.class);
+
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    private ClientConfiguration clientConfiguration;
+
+    public AccessTokenRetrieverServiceImpl(ClientConfiguration clientConfiguration) {
+        Assert.notNull(clientConfiguration);
+        this.clientConfiguration = clientConfiguration;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public OAuth2AccessToken getOAuth2AccessToken()
+        throws IOException, InvalidKeySpecException, NoSuchAlgorithmException {
+        String signedJwt = createSignedJWT(clientConfiguration);
+        String scope = String.join(" ", clientConfiguration.getScopes());
```

Comment:
What happens here if `scopes` are defined in env variables as a string.  does this throw any errors?

Reply:
+1, we need tests for empty string, comma-separated string, space-delimited, etc.
When in doubt follow what spring does when parsing env vars into arrays.

Reply:
Do we enforce a standard pattern for defining scopes in env variables as string? like comma separated or space separated?

Reply:
IMHO it depends on the usage, yaml would be a lit, a system property or environment variable would be a string.
as far as comma vs space, ideally support both, but one is required out of the box.  As much as i have opinions on this _should_ be formated, it's more important that the same format is used in the other SDKs 

@laura-rodriguez @bretterer, sorry if this is asked/commented on already, what is the input format for SCOPES?
```yaml
okta:
  client:
    scopes:
    - scopeA
    -scopeB
```
and for an ENV var:
```sh
export OKTA_CLIENT_SCOPES="scopeA, scopeB" # space optional?
```
(I don't think we have any array properties yet)

Reply:
Any inputs here, ppl?

Reply:
Sorry, i should have started this comment above, where the scopes are parsed, not where it is used.

Reply:
https://github.com/okta/okta-sdk-golang/blob/f273a5b89a1df4cfe52caec0c5f02899612b2d43/okta/requestExecutor.go#L144 tells me it is a space separated string array.

Reply:
Update: Got an answer that Unix env variable for scope is a space separated string.